app/femr/data/daos/system/PatientRepository.java

```diff
@@ -246,23 +246,32 @@ public IPatient retrievePatientById(Integer id) {
             Query<? extends IRankedPatientMatch> query = QueryProvider.getRankedPatientMatchQuery();
 
             DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
-            String ageString = dateFormat.format(new Date(age));
+            String ageString = "";
+            if(age != null) {
+                ageString = dateFormat.format(new Date(age));
+            }
+
 
             String sql
-                    = "select id, user_id, first_name, last_name, phone_number, age, sex, address, city, isDeleted, deleted_by_user_id, reason_deleted, ("+
-                    "case when phone_number = " + phone + " then 40 else 0 end + " +
-                    "case when last_name = \"" + lastName +"\" then 15 else 0 end + " +
+                    = "select id, user_id, first_name, last_name, phone_number, age, sex, address, city, isDeleted, deleted_by_user_id, reason_deleted, (";
+            if(phone != null && !phone.equals("")) {
+                sql += "case when phone_number = " + phone + " then 40 else 0 end + ";
+            }
+            sql += "case when last_name = \"" + lastName +"\" then 15 else 0 end + " +
                     "case when first_name = \"" + firstName +"\" then 10 else 0 end + " +
                     "case when dm_last_name = dm(\"" + lastName +"\") then 10 else 0 end + " +
-                    "case when dm_first_name = dm(\"" + firstName +"\") then 10 else 0 end + " +
-                    "case when address = \"" + addr +"\" then 15 else 0 end + " +
-                    "case when LEFT(first_name, 1)= \"" + firstName.substring(0, 1) +"\" then 10 else 0 end + " +
-                    "case when LEFT(last_name, 1)= \"" + lastName.substring(0, 1) +"\" then 10 else 0 end + " +
-                    "case when age != null and age = \"" + ageString + "\" then 10 else 0 end + " +
-                    "case when sex = \"" + gender + "\" then 10 else 0 end + " +
+                    "case when dm_first_name = dm(\"" + firstName +"\") then 10 else 0 end + ";
+            if(addr != null && !addr.equals("")) {
+                sql += "case when address = \"" + addr +"\" then 15 else 0 end + ";
+            }
+            if(age != null) {
+                sql += "case when age != null and age = \"" + ageString + "\" then 10 else 0 end + ";
+            }
+
+            sql += "case when sex = \"" + gender + "\" then 10 else 0 end + " +
```

Comment:
Why are there two null checks for age here? The one in the if statement and the one in the sql line on 268.